hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/CacheConfig.java

```diff
@@ -386,6 +399,17 @@ public boolean shouldReadBlockFromCache(BlockType blockType) {
     return false;
   }
 
+  /**
+   * Checks if the current heap usage is below the threshold configured by
+   * "hbase.rs.prefetchheapusage" (0.8 by default).
+   */
+  public boolean isHeapUsageBelowThreshold() {
+    double total = Runtime.getRuntime().maxMemory();
+    double available = Runtime.getRuntime().freeMemory();
+    double usedRatio = 1d - (available / total);
+    return heapUsageThreshold > usedRatio;
```

Comment:
[nit] I found the hbase has  of `MemorySizeUtil.safeGetHeapMemoryUsage();` in region server to get heap usage and other place with a single instance, although I think it's still using these two methods at the end.

Reply:
It doesn't seem to get accurate values, specially when running in the UTs.